tests/tests/test_configuration_deployments.py

```diff
@@ -132,3 +134,296 @@ def test_fail_duplicate_deployment(self, api_client_int, clean_db):
 
         rsp = requests.post(url, json=configuration_deployment, verify=False)
         assert rsp.status_code == 409
+
+
+class TestDevicesApiGetConfigurationDeploymentLink:
+    """ 
+        Tests /download/configuration/... download links.
+    """
+
+    @pytest.mark.parametrize(
+        "test_set",
+        [
+            {"dev_type": "rpi4", "name": "foo", "config": '{"foo":"bar"}'},
+            {"dev_type": "bb", "name": "bar", "config": '{"foo":"bar","baz":"qux"}'},
+        ],
+    )
+    def test_ok(self, api_client_int, clean_db, mongo, test_set):
+        """ 
+             Happy path - correct link obtained from the service, leading to a successful download
+             of a correct artifact.
+        """
+
+        # set up deployment
+        tenant_id = str(ObjectId())
+        _, r = api_client_int.create_tenant(tenant_id)
+        assert r.status_code == 201
+
+        deployment_id = str(uuid4())
+
+        dev = Device()
+        dev.device_type = test_set["dev_type"]
+
+        configuration_deployment = {
+            "name": test_set["name"],
+            "configuration": test_set["config"],
+        }
+
+        make_deployment(
+            api_client_int,
+            tenant_id,
+            deployment_id,
+            dev.devid,
+            configuration_deployment,
+        )
+
+        # obtain + verify deployment instructions
+        dc = SimpleDeviceClient()
+        nextdep = dc.get_next_deployment(
+            dev.fake_token_mt(tenant_id),
+            artifact_name="dontcare",
+            device_type=test_set["dev_type"],
+        )
+
+        assert nextdep.artifact["artifact_name"] == test_set["name"]
+        assert nextdep.artifact["source"]["uri"] is not None
+        assert nextdep.artifact["source"]["expire"] is not None
+        assert nextdep.artifact["device_types_compatible"] == [test_set["dev_type"]]
+
+        # get/verify download contents
+        r = requests.get(
+            nextdep.artifact["source"]["uri"],
+            headers={"Authorization": "Bearer " + dev.fake_token_mt(tenant_id)},
```

Comment:
Authorization header is usually not present when calling this URI.
```suggestion
```

Reply:
ok, I decided to remove Auth from all invocations.